### ISSUE TITLE
Add alternative setNonKeyToExcluded(), setNonPrimaryKeyToExcluded(), and setNonConflictingKeyToExcluded() methods to UPSERT syntax

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#16522]
This statement here:

```java
create().insertInto(TBook())
        .set(TBook_ID(), 5)
        .set(TBook_AUTHOR_ID(), 1)
        .set(TBook_PUBLISHED_IN(), 2000)
        .onDuplicateKeyUpdate()
        .setAllToExcluded()
        .execute();
```

Produces this query:

```sql
insert into "duckdb"."main"."t_book" ("ID", "AUTHOR_ID", "PUBLISHED_IN")
values (
  5, 
  1, 
  2000
)
on conflict ("ID")
do update
set
  "ID" = excluded."ID",
  "AUTHOR_ID" = excluded."AUTHOR_ID",
  "PUBLISHED_IN" = excluded."PUBLISHED_IN"
```

Which results in this exception on DuckDB:

```
org.jooq.exception.DataAccessException: SQL [insert into "duckdb"."main"."t_book" ("ID", "AUTHOR_ID", "PUBLISHED_IN") values (?, ?, ?) on conflict ("ID") do update set "ID" = excluded."ID", "AUTHOR_ID" = excluded."AUTHOR_ID", "PUBLISHED_IN" = excluded."PUBLISHED_IN"]; java.sql.SQLException: Binder Error: Can not assign to column 'ID' because it has a UNIQUE/PRIMARY KEY constraint
	at org.jooq_3.20.0-SNAPSHOT.DUCKDB.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3595)
	at org.jooq.impl.Tools.translate(Tools.java:3583)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:827)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:362)
	at org.jooq.impl.AbstractDelegatingQuery.execute(AbstractDelegatingQuery.java:115)
	at org.jooq.test.all.testcases.InsertUpdateTests.testInsertSetOnDuplicateKeyUpdateAllExcluded(InsertUpdateTests.java:3660)
	at org.jooq.test.jOOQAbstractTest.testInsertSetOnDuplicateKeyUpdateAllExcluded(jOOQAbstractTest.java:7347)
```

This is due to the limitations mentioned here:

- https://duckdb.org/docs/sql/indexes

In general, it doesn't make much sense to set the conflict constraint to itself, because that's effectively a no-op given that it's the conflict constraint.

This doesn't affect `ON DUPLICATE KEY UPDATE` when there are multiple constraints, in case of which the jOOQ query doesn't know which will be the effective conflict constraint per given row. This second bit hints at there possibly being other edge cases, to be further analysed.

The issue that added `setAllToExcluded()` was:

- https://github.com/jOOQ/jOOQ/issues/7912#issuecomment-1127576145

It also suggested:

- [x] `setNonKeyToExcluded()` would set all the values that are not part of any key. This is interesting in MySQL, where `ON DUPLICATE KEY` isn't just about the primary key but about any of the keys
- [x] `setNonPrimaryKeyToExcluded()` would set all the values that are not part of the primary key. This would allow for updating `UNIQUE` key values to a new value where this makes sense
- [x] `setNonConflictingKeyToExcluded()` would set all the values that are not part of the `ON CONFLICT` key. This would be PostgreSQL specific and allow for updating all the values of columns that are not the `ON CONFLICT` key.

The feature should be supported:

- [x] In the API
- [x] In the parser

**Comment by lukaseder** [comment]
Documentation task:

- https://github.com/jOOQ/jOOQ/issues/17621

**Comment by lukaseder** [comment]
Implemented in jOOQ 3.20
